stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+                    break;
+                }
+                _Offset = _New_offset;
+            }
+        }
+
+        if (_It != _Ls_vector.begin()) {
+            // Convert to the last leap second before or equal to _Time (except for the special case described above).
+            const auto _Last_leap       = &*(--_It);
+            const auto _Utc_leap_second = --utc_seconds{_Last_leap->date().time_since_epoch() + _Offset};
+#ifdef __cpp_lib_concepts
+            const auto _Leap_cmp = _Utc_leap_second <=> _Time_floor;
+#else // ^^^ __cpp_lib_concepts / TRANSITION, GH-395 workaround vvv
+            const auto _Leap_cmp = _Utc_leap_second > _Time_floor  ? strong_ordering::greater
+                                 : _Utc_leap_second == _Time_floor ? strong_ordering::equal
+                                                                   : strong_ordering::less;
+#endif // ^^^ workaround
+            if (_Tzdb._All_ls_positive && _STD is_gt(_Leap_cmp)) {
+                --_Offset;
+            }
+            return {_Last_leap->_Positive() && _STD is_eq(_Leap_cmp), _Offset};
+        } else {
+            return {false, _Offset};
+        }
+    }
+
+    class utc_clock {
+    public:
+        using rep                       = system_clock::rep;
+        using period                    = system_clock::period;
+        using duration                  = duration<rep, period>;
+        using time_point                = time_point<utc_clock>;
+        static constexpr bool is_steady = system_clock::is_steady;
+
+        _NODISCARD static time_point now() {
+            return from_sys(system_clock::now());
+        }
+
+        template <class _Duration>
+        _NODISCARD static sys_time<common_type_t<_Duration, seconds>> to_sys(const utc_time<_Duration>& _Utc_time) {
+            const auto _Lsi{get_leap_second_info(_Utc_time)};
+            sys_time<common_type_t<_Duration, seconds>> _Sys_time{_Utc_time.time_since_epoch() - _Lsi.elapsed};
+            if (_Lsi.is_leap_second) {
+                constexpr auto _Delta{seconds{1} - common_type_t<_Duration, seconds>{1}};
+                _Sys_time = _CHRONO floor<seconds>(_Sys_time) + _Delta;
+            }
+            return _Sys_time;
+        }
+
+        template <class _Duration>
+        _NODISCARD static utc_time<common_type_t<_Duration, seconds>> from_sys(const sys_time<_Duration>& _Sys_time) {
+            const auto& _Tzdb      = _CHRONO get_tzdb();
+            const auto& _Ls_vector = _Tzdb.leap_seconds;
+            const auto _It = _STD upper_bound(_Ls_vector.begin(), _Ls_vector.end(), _CHRONO floor<seconds>(_Sys_time));
+
+            seconds _Offset;
+            if (_Tzdb._All_ls_positive) {
+                _Offset = seconds{_It - _Ls_vector.begin()};
+            } else {
+                _Offset = seconds{0};
+                for (auto _Ls = _Tzdb.leap_seconds.begin(); _Ls != _It; ++_Ls) {
+                    _Offset += _Ls->value();
+                }

```

Comment:
Can't we make `leap_second` store the sum of all leap seconds before it, so we don't need to recalculate it every time?